File: ImageFiltersJava/app/src/main/java/org/geeksforgeeks/demo/MainActivity.java
package org.geeksforgeeks.demo;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.opengl.GLSurfaceView;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.mukesh.imageproccessing.OnProcessingCompletionListener;
import com.mukesh.imageproccessing.PhotoFilter;
import com.mukesh.imageproccessing.filters.*;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements OnProcessingCompletionListener, OnFilterClickListener {

    private PhotoFilter photoFilter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initialize();
    }

    private void initialize() {
        GLSurfaceView effectView = findViewById(R.id.effectView);
        RecyclerView effectsRecyclerView = findViewById(R.id.effectsRecyclerView);

        photoFilter = new PhotoFilter(effectView, this);
        photoFilter.applyEffect(
                BitmapFactory.decodeResource(getResources(), R.drawable.background),
                new None()
        );

        effectsRecyclerView.setLayoutManager(new LinearLayoutManager(this, RecyclerView.HORIZONTAL, false));
        effectsRecyclerView.setHasFixedSize(true);
        effectsRecyclerView.setAdapter(new Adapter(getItems(), this));
    }

    private List<Filters> getItems() {
        List<Filters> filterList = new ArrayList<>();
        filterList.add(new Filters("None", new None()));
        filterList.add(new Filters("AutoFix", new AutoFix()));
        filterList.add(new Filters("Highlight", new Highlight()));
        filterList.add(new Filters("Brightness", new Brightness()));
        filterList.add(new Filters("Contrast", new Contrast()));
        filterList.add(new Filters("Cross Process", new CrossProcess()));
        filterList.add(new Filters("Documentary", new Documentary()));
        filterList.add(new Filters("Duo Tone", new DuoTone()));
        filterList.add(new Filters("Fill Light", new FillLight()));
        filterList.add(new Filters("Fisheye", new FishEye()));
        filterList.add(new Filters("Flip Horizontally", new FlipHorizontally()));
        filterList.add(new Filters("Flip Vertically", new FlipVertically()));
        filterList.add(new Filters("Grain", new Grain()));
        filterList.add(new Filters("Grayscale", new Grayscale()));
        filterList.add(new Filters("Lomoish", new Lomoish()));
        filterList.add(new Filters("Negative", new Negative()));
        filterList.add(new Filters("Posterize", new Posterize()));
        filterList.add(new Filters("Rotate", new Rotate()));
        filterList.add(new Filters("Saturate", new Saturate()));
        filterList.add(new Filters("Sepia", new Sepia()));
        filterList.add(new Filters("Sharpen", new Sharpen()));
        filterList.add(new Filters("Temperature", new Temperature()));
        filterList.add(new Filters("Tint", new Tint()));
        filterList.add(new Filters("Vignette", new Vignette()));

        return filterList;
    }

    @Override
    public void onProcessingComplete(@NonNull Bitmap bitmap) {
    }

    @Override
    public void onFilterClicked(Filters filters) {
        if (photoFilter != null) {
            photoFilter.applyEffect(
                    BitmapFactory.decodeResource(getResources(), R.drawable.background),
                    filters.getFilter()
            );
        }
    }
}

File: ImageFiltersKotlin/app/src/main/java/org/geeksforgeeks/demo/MainActivity.kt
package org.geeksforgeeks.demo

import android.graphics.*
import android.opengl.GLSurfaceView
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.*
import com.mukesh.imageproccessing.*
import com.mukesh.imageproccessing.filters.*

class MainActivity : AppCompatActivity(), OnProcessingCompletionListener, OnFilterClickListener {
    private lateinit var result: Bitmap
    private var photoFilter: PhotoFilter? = null

    private lateinit var effectView: GLSurfaceView
    private lateinit var effectsRecyclerView: RecyclerView

    override fun onProcessingComplete(bitmap: Bitmap) {
        result = bitmap
    }

    override fun onFilterClicked(filters: Filters) {
        photoFilter?.applyEffect(
            BitmapFactory.decodeResource(resources, R.drawable.background), filters.filter
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initialize()
    }

    private fun initialize() {
        effectView = findViewById(R.id.effectView)
        effectsRecyclerView = findViewById(R.id.effectsRecyclerView)

        photoFilter = PhotoFilter(effectView, this)
        photoFilter?.applyEffect(
            BitmapFactory.decodeResource(resources, R.drawable.background),
            None()
        )
        effectsRecyclerView.layoutManager =
            LinearLayoutManager(this, RecyclerView.HORIZONTAL, false)
        effectsRecyclerView.setHasFixedSize(true)
        effectsRecyclerView.adapter = Adapter(getItems(), this)
    }

    private fun getItems(): MutableList<Filters> {
        return mutableListOf(
            Filters("None", None()),
            Filters("AutoFix", AutoFix()),
            Filters("Highlight", Highlight()),
            Filters("Brightness", Brightness()),
            Filters("Contrast", Contrast()),
            Filters("Cross Process", CrossProcess()),
            Filters("Documentary", Documentary()),
            Filters("Duo Tone", DuoTone()),
            Filters("Fill Light", FillLight()),
            Filters("Fisheye", FishEye()),
            Filters("Flip Horizontally", FlipHorizontally()),
            Filters("Flip Vertically", FlipVertically()),
            Filters("Grain", Grain()),
            Filters("Grayscale", Grayscale()),
            Filters("Lomoish", Lomoish()),
            Filters("Negative", Negative()),
            Filters("Posterize", Posterize()),
            Filters("Rotate", Rotate()),
            Filters("Saturate", Saturate()),
            Filters("Sepia", Sepia()),
            Filters("Sharpen", Sharpen()),
            Filters("Temperature", Temperature()),
            Filters("Tint", Tint()),
            Filters("Vignette", Vignette())
        )
    }
}